### PR TITLE
Add comments and xrefs

### PR DESCRIPTION
**Pull request by xwatkins**
Features
- [x] Molecule processing features
- [x] Amino acid modification features
- [x] Sites (features)

Free Text
- [x] Post-translational modification (free text?)

Keywords
- [x] Keywords -PTM

Xrefs - some missing, see https://www.ebi.ac.uk/panda/jira/browse/TRM-22369
- [x] Proteomic databases
- [x] 2D gel databases
- [x] PTM databases
- [x] Miscellaneous databases

### CODE REVIEW

File: src/model/uniprotkb/sections/ProteinProcessingConverter.ts [hunk 1]
```diff
@@ -53,5 +69,14 @@ export const convertProteinProcessing = (data: ProteinProcessingAPIModel) => {
     });
     proteinProcessingData.featuresData = features;
   }
+  if (data.databaseCrossReferences) {
+    const xrefs = getXrefsForSection(
+      data.databaseCrossReferences,
+      EntrySection.ProteinProcessing
+    );
+    if (xrefs && typeof xrefs !== 'undefined') {
```
Comment (dlrice) [inline src/model/uniprotkb/sections/ProteinProcessingConverter.ts]: i think `if (xrefs) {` will be sufficient

File: src/model/uniprotkb/sections/ProteinProcessingConverter.ts [hunk 2]
```diff
@@ -37,7 +46,14 @@ export const convertProteinProcessing = (data: ProteinProcessingAPIModel) => {
   const proteinProcessingData: ProteinProcessingUIModel = {
     featuresData: [],
     keywordData: [],
+    xrefData: [],
+    commentsData: [],
   };
+  if (data.comments) {
+    proteinProcessingData.commentsData = data.comments.filter(
+      d => d.commentType === Comment.PTM
```
Comment (dlrice) [inline src/model/uniprotkb/sections/ProteinProcessingConverter.ts]: can we make a var at the top of the file to hold valid comment types (`Comment.PTM`)

**Comment by dlrice** [comment]
There seems to be a large gap after the protvista feature viewer:
Screenshot 2019-06-11 at 15 48 54

**Comment by dlrice** [comment]
The old site has:
image

Will the new one has
image

Is that intentional?

**Comment by xwatkins** [comment]
At the moment it's displaying the headers based on the data coming from the backend, which is why some of the titles are slightly different. I want to have a chat with backend and Sangya in regards to this.

**Pull request merged**